src/Asset.php

```diff
@@ -24,6 +24,7 @@ interface Asset
     public const BLOCK_EDITOR_ASSETS = 32;
     public const BLOCK_ASSETS = 64;
     public const CUSTOMIZER_PREVIEW = 128;
+    public const ACTIVATE = 256;
```

Comment:
Do you think we should change the default location to `Assets::FRONTENT | Assets::ACTIVATE` here: https://github.com/inpsyde/assets/blob/34_asset_activate/src/BaseAsset.php#L110 ?

Reply:
Hm, this could be a breaking change when styles appear in a location they are not meant to be. But actually i never used any styles in `wp-activate.php`. Hm..i think we should leave it as it is right now. 🤔

Reply:
WordPress default behavior is that the default styles and scripts (wp_enqueue_script) are loaded. Also the default-header and footer and so on. Inpsyde Assets treating wp-activate.php as a different context means to add a the burden to the dev to always remember "i might need this also on wp-activate.php" while no one cares about wp-activate.php (until someone does). In reality this means wp-activate.php will not be styled by most themes utilizing Inpsyde Assets, while the default header/footer, html structure could easily be styled. By using FRONTEND and ACTIVATE as default we minimize this burden and have a constant reminder of ACTIVATE.

The current behavior could break something if someone utilized FRONTEND knowing wp-activate.php would not be affected. This in return means, this person wanted to style/script wp-activate.php differently, but this was something which could not be done with Inpsyde Assets in the first place. I think there was no way to register a script/style with Inpsyde Assets which would load on wp-activate.php. This to me limits the possible breaks enormously.

Reply:
WordPress default behavior is that the default styles and scripts (wp_enqueue_script) are loaded. Also the default-header and footer and so on. Inpsyde Assets treating wp-activate.php as a different context means to add a the burden to the dev to always remember "i might need this also on wp-activate.php" while no one cares about wp-activate.php (until someone does). In reality this means wp-activate.php will not be styled by most themes utilizing Inpsyde Assets, while the default header/footer, html structure could easily be styled. By using FRONTEND and ACTIVATE as default we minimize this burden and have a constant reminder of ACTIVATE.

The current behavior could break something if someone utilized FRONTEND knowing wp-activate.php would not be affected. This in return means, this person wanted to style/script wp-activate.php differently, but this was something which could not be done with Inpsyde Assets in the first place. I think there was no way to register a script/style with Inpsyde Assets which would load on wp-activate.php. This to me limits the possible breaks enormously.

Reply:
But that being said would mean, that we do not need `Asset::ACTIVATE` at all, because we can/should cover it via `Asset::FRONTEND` ? :)

Reply:
Was my initial idea, but I see the counter point to want to be able target wp-activate.php and rest of front end separately. So _if_ you want e.g. your default style not to load on wp-activate.php you can do so. But to me at this point you have already thought about the difference and you are aware of it, thus no problem of deviating from the default. I do not care too much about this separation tbh. What I care about is I can forget about wp-activate.php again and if the separation is wanted, I fight for `ACTIVATE` to be part of the default :)

Reply:
hmmm it's hard to differentiate. But we do similar on backend, where we use multiple hooks ("Locations") to register/enqueue assets but all are kind of used in Backoffice https://github.com/inpsyde/assets/blob/master/src/Util/AssetHookResolver.php#L63-L65 - so thats kind of similar to "Frontoffice" which also includes "wp-activate.php" (eventho it's not really related). I guess this is more or less a big inconsistency in WordPress (again). If we would use `Asset::FRONTEND` for Frontoffice _and_ `wp-activate.php`, then we could still do:

```
(new Style( ... ))->canEnqueue( fn(): bool => WpContext::determine()->isWpActivate() );
```

Reply:
Too me it is `FRONTEND` as too my understanding those contexts are WordPress specific. `LOGIN` is not really bound to login functionality (as depending on themes or plugins you can login from the `FRONTEND`) but to the WordPress specific wp-login.php, which has a completely different header/footer HTML-structure and hooks. So it is almost mandatory to serve different styles. In this sense wp-activate.php is `FRONTEND` (or a subcategory of `FRONTEND`). I think there is a difference how `inpsyde/assets` should treat those than `WpContext`.

Reply:
Eventho i don't like it, the default location in `Asset` is now `Asset::FRONTEND | Asset::ACTIVATE` - see here: https://github.com/inpsyde/assets/commit/6d70c645a539671dce2ee4d0e8d78bfa0f9447a7

Reply:
My thoughts were the following:

- `wp-activate.php` uses `get_header()` and `get_footer()` so it involves the current theme, so it make sense that when a theme enqueue assets those are used in `wp-activate.php`
- At the same time, `wp-activate.php` is not the "usual" theme-rendered page, because it has not any content and it has [hardcoded styles](https://github.com/WordPress/WordPress/blob/master/wp-activate.php#L107-L113) which might conflict with the theme. So it make sense to me that one might want to have assets _only_ for  `wp-activate.php`

In summary, I think that _most of the times_ people will want "frontend" assets for `wp-activate.php`, but also _sometimes_ they'll want assets only for `wp-activate.php`.

To satisfy both that conditions I proposed a `Asset::THEME` locations that would mean: _all the pages that call `get_header()` and `get_footer()`_<sup>**1**</sup>.

Right now, `Asset::THEME` would include what currently is , `Asset::FRONTEND` +  `wp-activate.php`. 
Because it is called `Asset::THEME`, it would be "natural" for people who write themes to use it. And making it the default location would automatically "upgrade" code that is using the current "frontend" default.

That said, people who want to target _only_  `wp-activate.php` could use `Assets::ACTIVATE` and people who would like to target all frontend _but_ `wp-activate.php` could use `Assets::FRONTEND`.

That would be the most flexible solution IMO giving the full choice to where to use assets. 

The current approach of using `Assets::ACTIVATE |  Assets::FRONTEND` (and have it as default) works pretty much the same way (after all, in my proposal `Assets::THEME = Assets::ACTIVATE |  Assets::FRONTEND`), however, it is very likely that people will forget to do that when registering theme assets.

---

<sup>**1**</sup> Which means that if at some point we realize there're other places in WP where `get_header()` and `get_footer()` are called we could add those to `Asset::THEME`

Reply:
I understand your reasoning and kind of agree with it, but i think that `Asset::THEME` is the wrong term. 
That's why i suggested to include it into `Asset::FRONTEND` - which is also wrong. `THEME` might fit, because you can overwrite those `theme-compat/header|footer[-*].php`-files in Theme - but this also matches to "embed", which is a query type.... 
Hm...a Theme is also loaded on `wp-login.php` or `/wp-admin/index.php`... i think we shouldn't bound the term "THEME" to `get_header|footer()`-calls or frontend-like alone. The term just sounds not right to me, but i cannot come up with a better term here, sry. :) 

I'm open for further discussions in a new ticket to "combine" or even add more cases which are might be missing right now. But let's not block this issue here, since the `Asset::ACTIVATE` is missing and was added. ✔️  🙃